site/app/controllers/SelfRejoinController.php

```diff
@@ -67,14 +67,13 @@ private function canRejoinCourse() {
     /**
      * @Route("/courses/{_semester}/{_course}/rejoin_course", methods={"POST"})
      *
-     * @return JsonResponse Successful response if the student meets the conditions
-     * to rejoin the course.
+     * @return RedirectResponse Course url if the student met the conditions to be readded.
      */
-    public function rejoinCourse(): JsonResponse {
+    public function rejoinCourse(): RedirectResponse {
+        $course_url = $this->core->buildCourseUrl();
+
         if (!$this->canRejoinCourse()) {
-            return JsonResponse::getFailResponse(
-                "This student does not meet the conditions to rejoin."
-            );
```

Comment:
You can do core->addErrorMessage() to give the client some idea of what happened.